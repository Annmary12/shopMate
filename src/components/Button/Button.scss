
@import '../../assets/scss/variables.scss';

%btn-placeholder {
  text-transform: capitalize;
  text-decoration: none;
  padding: 1rem 3rem;
  display: inline-block;
  border-radius: 2.4rem;
  font-family: ".SF NS Display";
  background-color: $white;
  transition: all .2s;
  font-size: 1.6rem;
  text-align: center;
  cursor: pointer;
  font-weight: 700;
  border: none;
}

.btn {
   &__primary {
    @extend %btn-placeholder;
    background: $primary-color;
    color: $white;
    border: none;

    &:hover {
      opacity: 0.8;
    }

    &:disabled {
      opacity: 0.5;
    }
   }

   &__secondary {
    @extend %btn-placeholder;
    color: $primary-color;

    &:hover {
      opacity: 0.8;
    }
   }
}

.medium {
  padding: 1.5rem 4rem;
}
