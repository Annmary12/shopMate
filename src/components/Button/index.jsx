// react libraries
import * as React from 'react';

// third-party labraries
import PropTypes from 'prop-types';

// style
import './Button.scss';

const Button = ({
  type,
  disabled,
  classes,
  name,
  size,
  onClick,
  isLoading = false
}) => (
  <button
    className={[classes +' '+ size]}
    type={type ? type : 'button'}
    disabled={disabled}
    onClick={onClick ? onClick : null}
  >
    { isLoading ? 'loading...' : name }
  </button>
)

Button.propTypes = {
  type: PropTypes.string,
  disabled: PropTypes.bool,
  classes: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  size: PropTypes.string,
  isLoading: PropTypes.bool,
};

export default Button;