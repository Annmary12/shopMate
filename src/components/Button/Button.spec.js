// react library
import React from 'react';

// third-party library
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

// component
import Button from './';

describe('Button component', () => {
  let wrapper;
  const props = {
    name: 'TestName',
    type: 'text',
    classes: 'TestClass'
  }

  beforeEach(() => {
   wrapper = mount(<Button {...props}/>)
  })
  it('should render button correctly', () => {
    const button = wrapper.find('button');

    expect(button.length).toBe(1);
    expect(button.props().type).toBe('text');
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})