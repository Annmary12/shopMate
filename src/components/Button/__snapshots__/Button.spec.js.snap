// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render button correctly 1`] = `
<Button
  classes="TestClass"
  name="TestName"
  type="text"
>
  <button
    className={
      Array [
        "TestClass undefined",
      ]
    }
    onClick={null}
    type="text"
  >
    TestName
  </button>
</Button>
`;
